addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/BodyGenerator.java

```diff
@@ -67,86 +63,127 @@ public BodyGenerator(Generators generators) {
                         }
                     });
 
-    public String generate(String name, boolean isArray, List<String> refs) {
+    public String generate(Schema schema) {
+        boolean isArray = schema instanceof ArraySchema;
         if (LOG.isDebugEnabled()) {
-            LOG.debug("Generate body for object " + name);
+            LOG.debug("Generate body for object " + schema.getName());
         }
-        String jsonStr = generateJsonObjectString(name, refs);
+
         if (isArray) {
-            jsonStr = createJsonArrayWith(jsonStr);
+            return generateFromArraySchema((ArraySchema) schema);
         }
-        return jsonStr;
-    }
 
-    public String generate(Property property, boolean isArray) {
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("Generate body for primitive type " + property.getType());
+        Map<String, Schema> properties = schema.getProperties();
+        if (properties != null) {
+            return generateFromObjectSchema(properties);
+        } else if (schema.getAdditionalProperties() instanceof Schema) {
+            return generate((Schema) schema.getAdditionalProperties());
         }
-        String jsonStr = generateJsonPrimitiveValue(property);
-        if (isArray) {
-            jsonStr = createJsonArrayWith(jsonStr);
+
+        if (schema instanceof ComposedSchema) {
+            return generateJsonPrimitiveValue(resolveComposedSchema((ComposedSchema) schema));
+        }
+        if (schema.getNot() != null) {
+            resolveNotSchema(schema);
+        }
+
+        // primitive type, or schema without properties
+        if (schema.getType() == null || schema.getType().equals("object")) schema.setType("string");
```

Comment:
While this isn't syntactically wrong, it would be best if conditionals use curly braces throughout the code, just for consistency/readability.

Note: I'm not hunting for these, the comment applies throughout (in case there are other occurrences I've overlooked).